base/prometheus/prometheus.Deployment.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    description: Collects metrics and aggregates them into graphs.
+  labels:
+    deploy: sourcegraph
+  name: prometheus
+spec:
+  minReadySeconds: 10
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: prometheus
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: prometheus
+    spec:
+      containers:
+      - image: sourcegraph/prometheus:3.8
+        env:
+        - name: USE_KUBERNETES_DISCOVERY
+          value: "true"
+        terminationMessagePolicy: FallbackToLogsOnError
+        name: prometheus
+        ports:
+        - containerPort: 9090
+          name: http
+        volumeMounts:
+        - mountPath: /prometheus
+          name: data
+        - mountPath: /sg_prometheus_add_ons
+          name: config
```

Comment:
It seems that the prometheus deployment also supports liveness and ready checks: 

- https://github.com/prometheus/prometheus/pull/2997
- https://github.com/helm/charts/blob/4126d8ddc08d832b2acb26490679a92e696e906e/stable/prometheus/templates/server-deployment.yaml#L97-L108 

Could you verify that these work (then add them to this yaml if so)?

Reply:
done, it works